Utils/Dataflow/009_oracleConnector/Oracle2JSON.py

```diff
@@ -0,0 +1,307 @@
+        'BTag': ['btagging'],
+        'Diboson': ['diboson', 'zz', 'ww', 'wz', 'wwbb', 'wwll'],
+        'DrellYan': ['drellyan', 'dy'],
+        'Exotic': ['exotic', 'monojet', 'blackhole', 'technicolor',
+                   'randallsundrum', 'wprime', 'zprime', 'magneticmonopole',
+                   'extradimensions', 'warpeded', 'randallsundrum',
+                   'contactinteraction', 'seesaw'],
+        'GammaJets': ['photon', 'diphoton'],
+        'Higgs': ['whiggs', 'zhiggs', 'mh125', 'higgs', 'vbf', 'smhiggs',
+                  'bsmhiggs', 'chargedhiggs'],
+        'Minbias': ['minbias'],
+        'Multijet': ['dijet', 'multijet', 'qcd'],
+        'Performance': ['performance'],
+        'SingleParticle': ['singleparticle'],
+        'SingleTop': ['singletop'],
+        'SUSY': ['bino', 'susy', 'pmssm', 'leptosusy', 'rpv', 'mssm'],
+        'Triboson': ['triplegaugecoupling', 'triboson', 'zzw', 'www'],
+        'TTbar': ['ttbar'],
+        'TTbarX': ['ttw', 'ttz', 'ttv', 'ttvv', '4top', 'ttww'],
+        'Upgrade': ['upgrad'],
+        'Wjets': ['w'],
+        'Zjets': ['z']}
+    match = {}
+    categories = []
+    for phys_category in PHYS_CATEGORIES_MAP:
+        current_map = [x.strip(' ').lower() for x in PHYS_CATEGORIES_MAP[phys_category]]
+        if hashtags is not None:
+            match[phys_category] = len([x for x in hashtags.lower().split(',') if x.strip(' ') in current_map])
+    categories = [cat for cat in match if match[cat] > 0]
+    if len(categories) > 0:
+        return categories
+    else:
+        if taskname is not None:
+            phys_short = taskname.split('.')[2].lower()
+            if re.search('singletop', phys_short) is not None: categories.append("SingleTop")
+            if re.search('ttbar', phys_short) is not None: categories.append("TTbar")
+            if re.search('jets', phys_short) is not None: categories.append("Multijet")
+            if re.search('h125', phys_short) is not None: categories.append("Higgs")
+            if re.search('ttbb', phys_short) is not None: categories.append("TTbarX")
+            if re.search('ttgamma', phys_short) is not None: categories.append("TTbarX")
+            if re.search('_tt_', phys_short) is not None: categories.append("TTbar")
+            if re.search('upsilon', phys_short) is not None: categories.append("BPhysics")
+            if re.search('tanb', phys_short) is not None: categories.append("SUSY")
+            if re.search('4topci', phys_short) is not None: categories.append("Exotic")
+            if re.search('xhh', phys_short) is not None: categories.append("Higgs")
+            if re.search('3top', phys_short) is not None: categories.append("TTbarX")
+            if re.search('_wt', phys_short) is not None: categories.append("SingleTop")
+            if re.search('_wwbb', phys_short) is not None: categories.append("SingleTop")
+            if re.search('_wenu_', phys_short) is not None: categories.append("Wjets")
+        return categories
```

Comment:
Missed default value if taskname is not None and all re.search returns None.

We have to check that categories length > 0

Reply:
It seems to me that it will return empty list in this case (defined at [267](https://github.com/PanDAWMS/dkb/pull/54/files/8fc1fe9e75c92877c4bdb16c043d9d85ce7fed69#diff-4c221825c8bb3980b06a244fe9499716R267)).
Have I missed something?

Reply:
Yes, empty list will be returned, but for unknown categories returns "Uncategorized".

Reply:
If it's OK - fine.
But why for tasks without taskname returns "Uncategorized"? It seems be same for any uncategorized tasks.